src/worker.rs

```diff
@@ -115,7 +115,17 @@ where
 
                     match self.write_with_retry(&record).await {
                         Ok(_) => {}
-                        Err(_) => continue,
+                        Err(e) => match e {
+                            Error::MaxRetriesExceeded => {
+                                error!("Reached MaxRetriesExceeded");
+                                break;
+                            }
+                            Error::ConnectionClosed => {
+                                error!("Reached ConnectionClosed");
```

Comment:
>crate users will be able to get access to the connection lost errors so they can do something about it.

I think we can't access this error because it isn't returned to the main task.
Are you assuming that users will take action based on the output logs?